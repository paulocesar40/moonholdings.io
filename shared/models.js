export const nomicsLink = 'https://nomics.com/';
