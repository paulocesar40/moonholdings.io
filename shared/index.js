export { moonHoldings, nomicsLink } from './copy';
