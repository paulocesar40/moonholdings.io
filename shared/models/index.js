export { moonHoldings, nomicsLink } from './copy'
export { defaultSquareStyle } from './coins'
