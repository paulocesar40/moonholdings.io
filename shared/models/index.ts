export * from './assets'
export * from './copy'
export * from './squares'
