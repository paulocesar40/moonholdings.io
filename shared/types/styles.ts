export interface ICoinStyle {
  color: string;
  backgroundColor: string;
  lightBg?: boolean;
}
