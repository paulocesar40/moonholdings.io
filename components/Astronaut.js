import React from 'react';

import { astronaut } from '../styles';

const { AstronautContainer, Heading } = astronaut;

export default ({ showLogo }) => (
  <AstronautContainer>
    { showLogo === true ? <Heading>MOON.HOLDINGS</Heading> : null }
    <img src="static/astronaut.png" alt="astronaut" />
  </AstronautContainer>
);
