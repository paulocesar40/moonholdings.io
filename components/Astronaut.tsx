import React from 'react';

import { moonHoldings } from '../shared/models';
import { astronaut } from '../styles';

const { AstronautContainer, Heading } = astronaut;

interface LogoCheck {
  showLogo: boolean;
}

export default (showLogo: LogoCheck) =>  (
  <AstronautContainer>
    { showLogo.showLogo === true ? <Heading>{moonHoldings}</Heading> : null }
    <img src="static/astronaut.png" alt="astronaut" />
  </AstronautContainer>
);
