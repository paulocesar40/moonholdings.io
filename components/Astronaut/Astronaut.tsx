import React from 'react'

import { moonHoldings } from '../../shared/models'
import { AstronautContainer, Heading } from '../../styles'

interface LogoCheck {
  showLogo: boolean;
}

export default (showLogo: LogoCheck) =>  (
  <AstronautContainer>
    { showLogo.showLogo === true ? <Heading>{moonHoldings}</Heading> : null }
    <img src="static/astronaut.png" alt="astronaut" />
  </AstronautContainer>
);
