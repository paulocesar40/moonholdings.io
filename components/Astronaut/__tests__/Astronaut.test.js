import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

// @ts-ignore (works with .tsx)
import Astronaut from '../Astronaut.tsx'
import { moonHoldings } from '../../../shared/models'
import { AstronautContainer, Heading } from '../../../styles'

describe('<Astronaut /> component', () => {
  describe('rendering', () => {
    const wrapper = shallow(<Astronaut showLogo={true} />);
    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
      expect(wrapper.contains(<AstronautContainer/>));
    });
  });

  describe('showLogo boolean', () => {
    it('should display the Heading if showLogo is true', () => {
      const wrapperShowLogoTrue = shallow(<Astronaut showLogo={true} />);
      expect(wrapperShowLogoTrue.containsAllMatchingElements([
        <Heading>{moonHoldings}</Heading>
      ])).toEqual(true);
    });

    it('should not display the Heading if showLogo is false', () => {
      const wrapperShowLogoFalse = shallow(<Astronaut showLogo={false} />);
      expect(wrapperShowLogoFalse.containsAllMatchingElements([
        <Heading>{moonHoldings}</Heading>
      ])).toEqual(false);
    });
  });
});
