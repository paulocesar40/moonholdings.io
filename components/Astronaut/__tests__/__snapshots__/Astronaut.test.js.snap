// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Astronaut /> component rendering should render a component matching the snapshot 1`] = `
<astronaut__AstronautContainer>
  <astronaut__Heading>
    MOON.HOLDINGS
  </astronaut__Heading>
  <img
    alt="astronaut"
    src="static/astronaut.png"
  />
</astronaut__AstronautContainer>
`;
